kolla/node_custom_config/doni.conf

```diff
@@ -1,5 +1,8 @@
 [DEFAULT]
-enabled_worker_types = ironic{% if enable_blazar | bool %},blazar.physical_host{% endif %}
+{# Note: newlines after these options are important so jinja doesn't squash them #}
+enabled_hardware_types = {% if enable_ironic | bool %}baremetal,{% endif %}{% if enable_k3s | bool %}device.balena,{% endif %}
```

Comment:
Having the logic in this template is getting very messy, and seems incorrect on this line. We need a better way to print a "list" of enabled drivers, without a chain of if statements.

Reply:
What part seems incorrect? It looks OK to me. Also for some reason this comment doesn't show up in the diff view 🤔 

Yeah it's not super readable. One long term architectural option would be to have workers disable themselves under some condition, e.g., if Blazar is not in the service catalogue, the Blazar worker never "turns on" when Doni starts its worker manager.

I'm not sure of a better way to template all that stuff though. Ansible tries to not give you too many control flow constructs to avoid templates being turing complete :) - that applies to stuff like tasks as well. It might require an additional filter, which seems like overkill.

Reply:
Oops, I selected the wrong line for the comment, misclick. Making a new comment with the correct line.